src/jit/codegencommon.cpp

```diff
@@ -4393,7 +4382,14 @@ void            CodeGen::genFnPrologCalleeRegArgs(regNumber xtraReg,
         else
         {
             // Since slot is typically 1, baseOffset is typically 0
-            int baseOffset = (regArgTab[argNum].slot - 1) * slotSize;
+            int slot = regArgTab[argNum].slot;
```

Comment:
Try this fix (at line 4359)
            if (varDsc->lvIsHfaRegArg())
            {
#ifdef _TARGET_ARM_
                // On ARM32 the storeType for HFA args is always TYP_FLOAT
                storeType = TYP_FLOAT;
                slotSize = (unsigned)emitActualTypeSize(storeType);
#else   // _TARGET_ARM64_
                storeType = genActualType(varDsc->GetHfaType());
                slotSize = (unsigned)emitActualTypeSize(storeType);
#endif // _TARGET_ARM64_
            }